core/src/main/java/org/elasticsearch/cluster/routing/UnassignedInfo.java

```diff
@@ -232,19 +232,27 @@ public long getLastComputedLeftDelayNanos() {
     }
 
     /**
-     * Updates delay left based on current time (in nanoseconds) and index/node settings.
+     * Calculates the delay left based on current time (in nanoseconds) and index/node settings.
      *
-     * @return updated delay in nanoseconds
+     * @return calculated delay in nanoseconds
      */
-    public long updateDelay(long nanoTimeNow, Settings settings, Settings indexSettings) {
-        long delayTimeoutNanos = getAllocationDelayTimeoutSettingNanos(settings, indexSettings);
-        final long newComputedLeftDelayNanos;
+    public long getRemainingDelay(long nanoTimeNow, Settings settings, Settings indexSettings) {
```

Comment:
nit: not sure of the preference, but I find it helpful to have method arguments `final` if they will not be changed.